src/altinn-app-frontend/src/utils/render.tsx

```diff
@@ -61,16 +61,24 @@ export function renderValidationMessages(
   id: string,
   variant: 'error' | 'warning' | 'info' | 'success',
 ) {
+  const key = JSON.stringify({ messages, id, variant });
+
   if (variant !== 'error') {
     return (
-      <SoftValidations variant={variant}>
+      <SoftValidations
+        variant={variant}
+        key={key}
```

Comment:
This seems odd to me - there is no looping over values here, so why should a key be nescessary?

I think the underlying problem here is the `renderValidationMessagesForComponent` code that returns an array of items directly, and that is returned, and used as a component (rendered). 

Shouldnt this be fixed instead by actually looping over the items and add a key there?


On a sidenote, shouldnt `id` be unique?

Reply:
Haha, yeah, you're right - you exposed my laziness.. :wink: I'll look into rewriting this to something better.

As for the `id` not being unique - yeah, I dunno.. I printed out all the `id` arguments given during the full test suite, and there were a bunch of duplicate IDs where `undefined` had been passed in (and just converted to a string at some point). It seems fickle and buggy to me, so I agree a proper cleanup is in order.

Reply:
I mean this "fixes" the problem, and definitely improves the output from tests - so maybe its worth merging this as-is now, and do a proper cleanup later. I assume that task is a bit bigger, and will require more testing. You have my approval at least 🙂

Reply:
Thanks! I'll look into it, but won't spend a week on it.. :wink: